nixos/modules/services/logging/packetbeat.nix

```diff
@@ -0,0 +1,185 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.packetbeat;
+
+  packetbeatYml = pkgs.writeText "packetbeat.yml" ''
+    name: ${cfg.name}
+    tags: ${builtins.toJSON cfg.tags}
+
+    ${cfg.configFlows}
+    ${cfg.configProtocols}
+    ${cfg.extraConfig}
+  '';
+
+in
+{
+  options = {
+
+    services.packetbeat = {
+
+      enable = mkEnableOption "packetbeat";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.packetbeat;
+        defaultText = "pkgs.packetbeat";
+        example = literalExample "pkgs.packetbeat7";
+        description = ''
+          The packetbeat package to use
+        '';
+      };
+
+      name = mkOption {
+        type = types.str;
+        default = "packetbeat";
+        description = "Name of the beat";
+      };
+
+      tags = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = "Tags to place on the shipped log messages";
+      };
+
+      stateDir = mkOption {
+        type = types.str;
+        default = "packetbeat";
+        description = ''
+          Directory below <literal>/var/lib/</literal> to store packetbeat's
+          own logs and other data. This directory will be created automatically
+          using systemd's StateDirectory mechanism.
+        '';
+      };
+
+      configFlows = mkOption {
```

Comment:
If this is just `yaml` does it even need is open option? [RFC42](https://github.com/NixOS/rfcs/pull/42) has been promoting a `settings` option which I think would benefit this module greatly.

Reply:
It is just yaml, this was a quick way of getting the module usable. My plan was to first get something usable, then refactor it to be smart too.